src/rust/id.shinigami/src/lib.rs

```diff
@@ -0,0 +1,287 @@
+            url: format!("{}/chapter/{}", BASE_URL, chapter_obj.get("chapter_id").as_string()?.read()),
+            date_updated: parse_date(chapter_obj.get("release_date").as_string()?.read()),
+            ..Default::default()
+        });
+    }
+
+    Ok(chapters)
+}
+
+#[get_manga_details]
+fn get_manga_details(id: String) -> Result<Manga> {
+    let url = format!("{}/v1/manga/detail/{}", API_URL, id);
+    let json = Request::new(url, HttpMethod::Get)
+        .header("Accept", "application/json")
+        .header("Origin", BASE_URL)
+        .json()?;
+
+    let data = json.as_object()?;
+    if data.get("retcode").as_int()? != 0 {
+        return Err(aidoku::error::AidokuError { 
+            reason: aidoku::error::AidokuErrorKind::NodeError(aidoku::error::NodeError::ParseError) 
+        });
+    }
+
+    let manga_obj = data.get("data").as_object()?;
+    let title = manga_obj.get("title").as_string()?.read();
+    let _alt_title = manga_obj.get("alternative_title").as_string()?.read();
+    
+    let cover = if let Ok(url) = manga_obj.get("cover_portrait_url").as_string() {
+        if !url.read().is_empty() {
+            url.read()
+        } else {
+            manga_obj.get("cover_image_url").as_string()?.read()
+        }
+    } else {
+        manga_obj.get("cover_image_url").as_string()?.read()
+    };
+
+    let taxonomy = manga_obj.get("taxonomy").as_object()?;
+    let mut genres = Vec::new();
+    if let Ok(genre_array) = taxonomy.get("Genre").as_array() {
+        for genre in genre_array {
+            if let Ok(genre_obj) = genre.as_object() {
+                if let Ok(genre_name) = genre_obj.get("name").as_string() {
+                    genres.push(genre_name.read());
+                }
+            }
+        }
+    }
+
+    // Fix in get_manga_details function
+    Ok(Manga {
+        id: manga_obj.get("manga_id").as_string()?.read(),
+        title,
+        author: get_taxonomy_names(&taxonomy, "Author"),
+        artist: get_taxonomy_names(&taxonomy, "Artist"),
+        description: manga_obj.get("description").as_string()?.read(),
+        url: manga_obj.get("manga_id").as_string()?.read(),
+        cover,
+        categories: genres,
+        status: match manga_obj.get("status").as_int()? {
+            1 => MangaStatus::Ongoing,
+            2 => MangaStatus::Completed,
+            _ => MangaStatus::Unknown,
+        },
+        nsfw: MangaContentRating::Safe,
+        viewer: MangaViewer::Rtl
+        // Remove ..Default::default() here
+    })
+}
+
+#[modify_image_request]
+fn modify_image_request(request: Request) {
+    request
+        .header("Accept", "image/avif,image/webp,image/apng,image/svg+xml,image/*,*/*;q=0.8")
+        .header("DNT", "1")
+        .header("Referer", BASE_URL)  // Changed from format!() to direct string
+        .header("Sec-Fetch-Dest", "empty")
+        .header("Sec-GPC", "1");
+}
+
+// Helper function to parse date
+fn parse_date(date_str: String) -> f64 {
```

Comment:
it would be cleaner if you moved helper functions (since you have multiple) into a `helpers.rs` module.

also, while this seems like it's probably fine, you could alternatively use the built in date parsing from aidoku-rs std (`as_date`) to parse a date string with a given format into epoch time